src/server/routes/v1.ts

```diff
@@ -332,7 +309,34 @@ export default async (fastify: FastifyInstance): Promise<void> => {
     return `updated ${updatedYamls.length} packages at ${new Date().toISOString()}`;
   });
 
+  // *----------------- override package image ---------------------
+  fastify.post("/ghs/imageOverride", async (request, reply) => {
+    const accessToken = request.headers["xxx-access-token"];
+    if (accessToken == null) {
+      reply.status(401);
+      return new Error("unauthorised");
+    }
+    if (accessToken !== API_ACCESS_TOKEN) {
+      reply.status(403);
+      return new Error("forbidden");
+    }
+
+    const packageService = new PackageService();
+
+    const { pkgId, iconUrl } = request.body;
+
+    const pkgExist = await packageService.findOne({ filters: { Id: pkgId as string } });
+
+    if (pkgExist == null || iconUrl === "") {
+      return "package not found, or bad rquest";
+    }
+
+    pkgExist.IconUrl = iconUrl;
```

Comment:
this wont actually save the thing in the database, what i recommend is either findOne, check if it exists, and then update, or just run an update (and not updateOne cos we probably wanna change the icon on all versions), and check how many were actually updated (returned by the function as updatedCount or something that, just check the return type)